### PR TITLE
feat(contributing): addition of a pre commit githook (formatting) and contributing doc

### PR DESCRIPTION
**Pull request by dimakis**
### Description
Describe your changes in detail
Addition of a pre-commit `format` check githook which uses go formatting tools under the hood. 
Also addition of the commands to set it up for the project in the makefile and explained usage in newly created CONTRIBUTING.md doc.

### Verification
* Run `make setup/git/hooks` 
* Try and commit a go file with incorrect formatting
* Should get a response to run `make format` to fix

### Type of change

~- [ ] Bug fix (non-breaking change which fixes an issue)~
- [x] New feature (non-breaking change which adds functionality)
~- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)~
~- [ ] Documentation change~
~- [ ] Other (please specify)~

### Checklist

- [x] Documentation added for the feature
~- [ ] CI and all relevant tests are passing~
- [ ] Code Review completed
- [ ] Verified independently by reviewer

## Screenshots (if appropriate):

### CODE REVIEW

File: .githooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+staged_go_files=$(git diff --cached --name-only | grep ".go$")
```
Comment (wtrocki) [inline .githooks/pre-commit:3]: Using make targets would be better but good for now

**Review by wtrocki — APPROVED** [review]

File: .githooks/pre-commit (hunk 1 above)
Comment (wtrocki) [reply .githooks/pre-commit:3]: What I mean is that we should have go vet make target etc.

**Comment by jjkiely** [comment]
Verified that it's working 
image

**Comment by jjkiely** [comment]
Good stuff @dimakis

File: .githooks/pre-commit (hunk 1 above)
Comment (dimakis) [reply .githooks/pre-commit:3]: a quick win as I'm seeing alot of PRs failing for formatting errors. I'm not familiar with make targets. will have a look into them at some stage

**Pull request merged**